app/services/testdata/InProgressQuestionnaireStatusGenerator.scala

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright 2017 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package services.testdata
+
+import model.persisted.{ QuestionnaireAnswer, QuestionnaireQuestion }
+import model.testdata.CreateCandidateData
+import play.api.mvc.RequestHeader
+import repositories._
+import repositories.application.GeneralApplicationRepository
+import services.testdata.candidate.{ ConstructiveGenerator, InProgressAssistanceDetailsStatusGenerator }
+import services.testdata.faker.DataFaker._
+import uk.gov.hmrc.http.HeaderCarrier
+import scala.concurrent.ExecutionContext.Implicits.global
+
+object InProgressQuestionnaireStatusGenerator extends InProgressQuestionnaireStatusGenerator {
+  override val previousStatusGenerator = InProgressAssistanceDetailsStatusGenerator
+  override val appRepository = applicationRepository
+  override val qRepository = questionnaireRepository
+
+}
+
+trait InProgressQuestionnaireStatusGenerator extends ConstructiveGenerator {
+  val appRepository: GeneralApplicationRepository
+  val qRepository: QuestionnaireRepository
+
+  // scalastyle:off method.length
```

Comment:
I think the nested `getXXX` functions you have can be private functions(or just public, they seem useful) in the trait and we won't need to turn off this checker

Reply:
👍